xtext/org.icyphy.linguafranca/src/org/icyphy/generator/Main.xtend

```diff
@@ -81,47 +115,51 @@ class Main {
         }
         props
     }
-
-	def protected runGenerator(String string, Properties properties) {
-		// Load the resource
-		val set = resourceSetProvider.get
-		val fileRoot = (new File("")).getAbsolutePath()
-		val fileName = fileRoot + File.separator + string;
+    
+    /**
+     * Load resources, read in the code, parse it, validate it, and, finally,
+     * invoke the code generator.
+     */
+    def protected runGenerator(String string, Properties properties) {
+        // Load the resource
+        val set = resourceSetProvider.get
+        val fileRoot = (new File("")).getAbsolutePath()
+        val fileName = fileRoot + File.separator + string;
 
         val f = new File(fileName)
         if (!f.exists) {
-            System::err.println('lfc: error: ' + fileName + ': No such file or directory');
+            System::err.println('lfc: error: ' + fileName +
+                ': No such file or directory');
             throw new FileNotFoundException(fileName);
         }
-        
-		val resource = set.getResource(URI.createFileURI(fileName), true)
-
-		// Read the code
-		val code = new StringBuilder();
-		
-			val reader = new BufferedReader(new FileReader(fileName));
-			var String line;
-			while ((line = reader.readLine()) !== null) {
-				code.append(line).append("\n");
-			}
-		
-
-		// Validate the resource
-		val issues = validator.validate(resource, CheckMode.ALL, CancelIndicator.NullImpl)
-		if (!issues.empty) {
-			System::err.println('Aborting. Unable to validate resource.');
-			issues.forEach[System.err.println(it)]
-			System::exit(1)
-		}
-
-		// Configure and start the generator
-		fileAccess.outputPath = 'src-gen'
-		val context = new StandaloneContext => [
-			cancelIndicator = CancelIndicator.NullImpl
-			args = properties;
-		]
-		
-		generator.generate(resource, fileAccess, context)
-		System.out.println('Code generation finished.')
-	}
+
+        val resource = set.getResource(URI.createFileURI(fileName), true)
+
+        // Read the code
+        val code = new StringBuilder();
```

Comment:
Looks like the `code` variable is unused in code generation.

Reply:
Ah, you're right! Looks like an old remnant that is obsolete. I'll remove it.